net/initiate.go

```diff
@@ -107,8 +107,10 @@ func (h *Host) handleProtocolStream(stream libp2pnetwork.Stream) {
 	}
 
 	if err := p2pnet.WriteStreamMessage(stream, resp, stream.Conn().RemotePeer()); err != nil {
-		log.Warnf("failed to send response to peer: err=%s", err)
-		_ = s.Exit()
+		log.Warnf("failed to send response to peer: %s", err)
+		if err = s.Exit(); err != nil {
+			log.Warnf("Swap exit failure: %s", err)
+		}
```

Comment:
For the most part, the PR stays on theme, but I added logging to the 3 places that were swallowing the bubbled up error from calls to the swap `Exit()` method. Without inspecting the error, we can't judge how critical the error is. There were places in previous PRs, where it was decided not to log critical errors because they were getting bubbled up. This means we need to ensure that we never silently swallow a bubbled-up error.